timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -178,6 +179,87 @@ void declare_context(py::module &m) {
             Coordinates have shape (F, N, 3)
             Boxes have shape (F, 3, 3)
 
+    )pbdoc")
+        .def(
+            "multiple_steps_local",
+            [](timemachine::Context &ctxt,
+               const int n_steps,
+               const py::array_t<int, py::array::c_style> &local_idxs,
+               const int store_x_interval,
+               const double radius,
+               const double k,
+               const double temperature,
+               const int seed) -> py::tuple {
+                if (n_steps <= 0) {
+                    throw std::runtime_error("local steps must be at least one");
+                }
+
+                const int x_interval = (store_x_interval <= 0) ? n_steps : store_x_interval;
+
+                std::vector<int> vec_local_idxs(local_idxs.size());
+                std::memcpy(vec_local_idxs.data(), local_idxs.data(), vec_local_idxs.size() * sizeof(int));
+                // Verify that local idxs are unique
+                unique_idxs<int>(vec_local_idxs);
+                std::array<std::vector<double>, 2> result =
+                    ctxt.multiple_steps_local(n_steps, vec_local_idxs, x_interval, radius, k, temperature, seed);
+
+                const int N = ctxt.num_atoms();
+                const int D = 3;
+                const int F = result[0].size() / (N * D);
+                py::array_t<double, py::array::c_style> out_x_buffer({F, N, D});
+                std::memcpy(out_x_buffer.mutable_data(), result[0].data(), result[0].size() * sizeof(double));
+
+                py::array_t<double, py::array::c_style> box_buffer({F, D, D});
+                std::memcpy(box_buffer.mutable_data(), result[1].data(), result[1].size() * sizeof(double));
+                return py::make_tuple(out_x_buffer, box_buffer);
+            },
+            py::arg("n_steps"),
+            py::arg("local_idxs"),
+            py::arg("store_x_interval") = 0,
+            py::arg("radius") = 1.2,
+            py::arg("k") = 10000.0,
+            py::arg("temperature") = 300.0,
+            py::arg("seed") = 2022,
+            R"pbdoc(
+        Take multiple steps using particles selected based on the log probability using a random particle from the local_idxs,
+        the random particle is frozen for all steps.
+
+        Running a barostat and local MD at the same time are not currently supported. If a barostat is
+        assigned to the context, the barostat won't run.
+
+        F = iterations / store_x_interval
+
+        Parameters
+        ----------
+        n_steps: int
+            Number of steps to run.
+
+        local_idxs: np.array of int32
+            The idxs that defines the atoms to use as the region(s) to run local MD. A random idx will be
+            selected to be frozen and used as the center of the shell of particles to be simulated.
```

Comment:
Do we select a different random index at every step, or only once at the beginning? (seems helpful to specify here)

Reply:
Specified here: https://github.com/proteneer/timemachine/pull/915/commits/c13e0e4522e9d7f52f681291347289a6a66c81bf